src/liboslexec/runtimeoptimize.cpp

```diff
@@ -2121,6 +2122,26 @@ RuntimeOptimizer::optimize_ops (int beginop, int endop,
         // Nothing below here to do for no-ops, take early out.
         if (op->opname() == u_nop)
             continue;
+        // If an exit or return was reached, kill the remaining instructions
+        if (optimize() && (op->opname() == u_return || op->opname() == u_exit)) {
+            ASSERT(skipops == 0 && "Unexpected skipops");
+            if (m_in_conditional[opnum]) {
+                // Nop the remaining instructions in the same basic block
+                while (next_block_instruction(opnum)) {
```

Comment:
Now that you fixed it so that it recognizes exit as the end of a basic block, does this code actually do anything? Seems like the `while` loop will never execute its body since the `exit` instruction is now -- correctly -- the last instruction in its basic block.

Reply:
Yes, I think that should be `while (m_in_conditional[++opnum])`